<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Parameters - MODAS</title>
  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Parameters";
    var mkdocs_page_input_path = "parameters.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> MODAS</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../installation/">Installation</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../quickstart/">QuickStart</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../tutorial/">Tutorial</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Parameters</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#genotype-dimensionality-reduction-module-genoidx">Genotype Dimensionality Reduction Module (genoidx)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#phenotype-preprocessing-and-transformation-module-phenorm">Phenotype preprocessing and transformation module (phenorm)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#omics-phenotype-pre-screening-module-prescreen">Omics Phenotype Pre-screening Module (prescreen)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#regional-gwas-identify-qtl-module-regiongwas">Regional gwas identify QTL module (regiongwas)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#mendelian-randomization-analysis-modulemr">Mendelian randomization analysis module(mr)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#whole-genome-wide-association-analysis-and-visualization-modulevisual">Whole genome-wide association analysis and visualization module(visual)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#construction-of-stress-response-index-and-identification-of-stress-responsive-qtlcontrast">Construction of stress response index and identification of stress-responsive QTL(contrast)</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#co-localization-analysis-of-qtls-for-multiple-traitscoloc">Co-localization analysis of QTLs for multiple traits(coloc)</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../visual_example/">Visual_example</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../modas_for_windows_mac/">MODAS for Windows/Mac</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">MODAS</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Parameters</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="parameters">Parameters</h1>
<p>This chapter introduces the parameters of each module in detail. </p>
<h2 id="genotype-dimensionality-reduction-module-genoidx">Genotype Dimensionality Reduction Module (genoidx)</h2>
<p><code>-g</code>: Genotype file used for genome-wide genotype dimensionality reduction analysis, support plink bed format and hapmap formatï¼Œthe plink bed format file only needs to provide the file prefix, the hapmap format needs to be used together with the parameter <code>-convert</code><br/>
<code>-genome_cluster</code>: Perform genome-wide genotype dimensionality reduction analysis on genotype file<br/> 
<code>-w</code>: Window size for genome-wide genotype dimensionality reduction analysis(bp), default 1000000<br/>
<code>-s</code>: Step size for genome-wide genotype dimensionality reduction analysis(bp), default 500000<br/>
<code>-convert</code>: Convert hapmap genotype format to plink bed format genotype<br/>
<code>-clump</code>: Clumping analysis for genotype file,  used to keep a subset of SNPs that are nearly uncorrelated with each other, thereby reducing the number of total SNPs to speed up genome-wide genotype dimensionality reduction analysis <br/>
<code>-p</code>: Number of threads for genome cluster analysis,max threads is total number of genome chromosomes<br/>
<code>-o</code>: The prefix of the output file, default MODAS_genoidx_output</p>
<h2 id="phenotype-preprocessing-and-transformation-module-phenorm">Phenotype preprocessing and transformation module (phenorm)</h2>
<p><code>-phe</code>: Phenotype file for phenotype preprocessing and transformation, the file format is csv format, the first column and the first row are the names of inbred lines and phenotypes respectively<br/>
<code>-g</code>: Genotype file in plink bed format for principal component analysis, used with the parameter <code>-pca</code> to correct differences in phenotype caused by population structure<br/>
<code>-pca</code>: Correct the differences in phenotype caused by population structure through PCA<br/>
<code>-r</code>: Phenotype missing ratio filter cutoff<br/>
<code>-v</code>: Phenotype value filter cutoff<br/>
<code>-log2</code>: log2 transformation of phenotype<br/>
<code>-log10</code>: log10 transformation of phenotype<br/>
<code>-ln</code>: ln transformation of phenotype<br/>
<code>-qqnorm</code>: Normal quantile transformation of phenotype<br/>
<code>-norm</code>: Boxcox transformation of phenotype<br/>
<code>-o</code>: The prefix of the output file, default MODAS_phenorm_output<br/></p>
<h2 id="omics-phenotype-pre-screening-module-prescreen">Omics Phenotype Pre-screening Module (prescreen)</h2>
<p><code>-g</code>: genotype file in plink bed format, used to calculate the kinship matrix for GWAS analysis<br/>
<code>-genome_cluster</code>: pseudo-genotype file for phenotype pre-screening, generated by subcommand <code>genoidx</code><br/>
<code>-phe</code>: phenotype file for phenotype pre-screening analysis, the file format is csv format, the first column and the first row are the names of inbred lines and phenotypes respectively <br/>
<code>-lm_suggest_pvalue</code>: linear model suggest pvalue for pseudo-SNP filter used for association analysis in mixed linear model, default 1/ pseudo-genotype file pseudo-snp number<br/>
<code>-lmm_suggest_pvalue</code>: lmm suggest pvalue for phenotype prescreen, used to obtain candidate phenotypes with significant association analysis signals and significant association regions of phenotypes, default 1e-6<br/>
<code>-p</code>: number of threads for analysis in prescreen sub command<br/>
<code>-o</code>: The prefix of the output file, default MODAS_prescreen_output</p>
<h2 id="regional-gwas-identify-qtl-module-regiongwas">Regional gwas identify QTL module  (regiongwas)</h2>
<p><code>-g</code>: Genotype file in plink bed format, used to calculate the kinship matrix for regional gwas analysis and extract snp in significant association regions of phenotypes <br/>
<code>-phe</code>: Candidate phenotype file generated by subcommand <code>prescreen</code> <br/>
<code>-phe_sig_qtl</code>: Significant association regions of candidate phenotypes file generated by subcommand <code>prescreen</code><br/>
<code>-p1</code>: Significance threshold for index SNPs, used to determine the phenotype with QTL and index snps of phenotype<br/>
<code>-p2</code>: Secondary significance threshold for clumped SNPs, Used to obtain the secondary significant SNP linked to index snp to determine the reliability of the QTL. MODAS outputs the QTL containing 10 secondary significant SNPs as the phenotypic QTL result<br/>
<code>-cluster</code>: Cluster phenotype by QTL<br/>
<code>-p</code>: Number of threads for regional gwas analysis, default 1<br/>
<code>-o</code>: The prefix of the output file, default MODAS_regiongwas_out<br/></p>
<h2 id="mendelian-randomization-analysis-modulemr">Mendelian randomization analysis module(mr)</h2>
<p><code>-g</code>: Genotype file in plink bed format<br/>
<code>-exposure</code>: Exposure phenotype file, such as mTrait phenotype, the file format is csv format, the first column and the first row are the names of inbred lines and phenotypes respectively<br/>
<code>-outcome</code>: Outcome phenotype file, such as pTrait phenotype,  the file format is csv format, the first column and the first row are the names of inbred lines and phenotypes respectively<br/>
<code>-qtl</code>: Exposure phenotype QTL file, generated by subcommand <code>regiongwas</code><br/>
<code>-lm</code>: Perform Mendelian Randomization through linear model<br/>
<code>-mlm</code>: Perform Mendelian Randomization through mixed linear model<br/>
<code>-p</code>: Number of threads for Mendelian Randomization analysis<br/>
<code>-pvalue</code>: The pvalue cutoff of Mendelian randomization analysis result output,default 1 <br/>
<code>-net</code>: Mendelian Randomization network analysis, used to identify functional modules<br/>
<code>-o</code>: The prefix of the output file, default MODAS_mr_out</p>
<h2 id="whole-genome-wide-association-analysis-and-visualization-modulevisual">Whole genome-wide association analysis and visualization module(visual)</h2>
<p><code>-g</code>: Genotype file in plink bed format for whole genome-wide association analysis<br/>
<code>-phe</code>: phenotype file for GWAS analysis and visualization<br/>
<code>-gwas_model</code>: model for GWAS analysis, supports the models provided by gemma, rMVP, and GAPIT software, including gemma_LM, gemma_MLM, rMVP_GLM, rMVP_MLM, rMVP_FarmCPU, GAPIT_GLM, GAPIT_MLM, GAPIT_CMLM, GAPIT_MLMM, GAPIT_FarmCPU, GAPIT_Blink. Default is gemma_MLM<br/>
<code>-qtl</code>: phenotype QTL file generated by subcommand <code>regiongwas</code><br/>
<code>-visual</code>: Generate web-based GWAS visualization results<br/>
<code>-anno</code>: gene annotation file for QTL annotation, used to display QTL information in visualized results<br/>
<code>-p</code>: Number of threads genome-wide association analysis and visualization<br/>
<code>-o</code>: The prefix of the output file,default MODAS_visual_out</p>
<h2 id="construction-of-stress-response-index-and-identification-of-stress-responsive-qtlcontrast">Construction of stress response index and identification of stress-responsive QTL(contrast)</h2>
<p><code>-stress_phe</code>: omics dataset under stress treatment<br/>
<code>-control_phe</code>: omics dataset under control treatment<br/>
<code>-alpha</code>: Contrast strength for contrastive PCA, it represents the trade-off between having the high stress variance and the low control variance, default 1<br/>
<code>-n_comp</code>: Number of components for contrastive PCA, default 2<br/>
<code>-beta_test_pvalue</code>: Significance threshold for stress-responsive CPCA PC, used to determine the CPCA PC representing stress response effect of molecular features, default 1/number of CPCA PC for beta test<br/>
<code>-gwas</code>: perform GWAS on stress response index to identify stress-responsive molecular QTL<br/>
<code>-g</code>: Genotype file in plink bed format, used to perform GWAS on stress response index<br/>
<code>-genome_cluster</code>: Pseudo-genotype file for phenotype pre-screening, generated by subcommand genoidx<br/>
<code>-anova_pvalue</code>: Significance threshold of two-way ANOVA, used to determine stress-responsvie molecular QTL, default 0.05<br/>
<code>-gwas_model</code>: GWAS model for association analysis, supporting LM, GLM and MLM, default MLM<br/>
<code>-p</code>: Number of threads for gwas analysis, default 1<br/>
<code>-o</code>: The prefix of the output file, default MODAS_contrast_out</p>
<h2 id="co-localization-analysis-of-qtls-for-multiple-traitscoloc">Co-localization analysis of QTLs for multiple traits(coloc)</h2>
<p><code>-qtl</code>: molecular QTL file for co-localization analysis<br/>
<code>-g</code>: Genotype file in plink bed format, used to extract genotype of SNPs associated with molecular features<br/>
<code>-gwas_dir</code>: directory containing GWAS results for molecular features, used to extract SNPs associated with molecular features. It is recommended to use the association analysis results generated by the MODAS subcommand <code>regiongwas</code><br/>
<code>-metric</code>: metric for determining clustering patterns of kinship, it can be calinski_harabasz or silhouette, defualt calinski_harabasz<br/>
<code>-eps</code>: The maximum distance between two traits used to consider a trait to co-localize with the other trait, defualt 0.2<br/>
<code>-pvalue</code>: Threshold of GWAS results for screening SNPs associated with molecular features, defualt 1e-6<br/>
<code>-p</code>: Number of threads for co-localization analysis, default 1
<code>-o</code>: The prefix of the output file, default MODAS_coloc_out</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../visual_example/" class="btn btn-neutral float-right" title="Visual_example">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../tutorial/" class="btn btn-neutral" title="Tutorial"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../tutorial/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../visual_example/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../javascripts/config.js" defer></script>
      <script src="../javascripts/mathjax.js" defer></script>
      <script src="https://polyfill.io/v3/polyfill.min.js?features=es6" defer></script>
      <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
