<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>QuickStart - MODAS</title>
  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "QuickStart";
    var mkdocs_page_input_path = "quickstart.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> MODAS</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../installation/">Installation</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">QuickStart</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#input-format">Input format</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#genotype-format">Genotype format</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#phenotype-format">Phenotype format</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#new-features-in-modas2">New features in MODAS2</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#identifying-stress-responsive-molecular-qtl-through-contrastive-pca-and-two-way-anova">Identifying stress-responsive Molecular QTL through contrastive PCA and Two-Way ANOVA</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#multi-trait-qtl-colocalization-analysis-based-on-image-matching-algorithms">Multi-Trait QTL Colocalization Analysis Based on Image Matching Algorithms</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#modules-in-modas">Modules in MODAS</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#generate-pseudo-genotype-files">Generate pseudo-genotype files</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#prescreen-candidate-genomic-regions-for-omics-data">Prescreen candidate genomic regions for omics data</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#perform-regional-association-analysis-to-identify-qtls">Perform regional association analysis to identify QTLs</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#perform-mendelian-randomization-analysis">Perform Mendelian randomization analysis</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#whole-genome-wide-association-analysis-and-visualization">Whole genome-wide association analysis and visualization</a>
    </li>
        </ul>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../tutorial/">Tutorial</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../parameters/">Parameters</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../visual_example/">Visual_example</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../modas_for_windows_mac/">MODAS for Windows/Mac</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">MODAS</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>QuickStart</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="quick-start">Quick Start</h1>
<p>The basic use of MODAS generally includes three steps: first, generate pseudo-genotype files; second, use the pseudo-genotype files to screen candidate genomic blocks; third, identify QTLs by SNP-based regional association analysis. In addition, Mendelian randomization (MR) analysis and GWAS visualization can also be performed for specified traits. In this part, only basic commands are listed and briefly described for a quick start, please refer to the tutorial part for a more detailed description of the implementation principle and parameters of MODAS.</p>
<h2 id="input-format">Input format</h2>
<h3 id="genotype-format">Genotype format</h3>
<p>MODAS takes genotype files in plink-bed format as input, it also supports the conversion of hapmap format to plink-bed format. For usage, refer to the subcommand <code>genoidx</code>.</p>
<h3 id="phenotype-format">Phenotype format</h3>
<p>MODAS takes omics phenotype files in csv format as input, the first column and the first row are the names of inbred lines and phenotypes respectively.</p>
<h2 id="new-features-in-modas2">New features in MODAS2</h2>
<h3 id="identifying-stress-responsive-molecular-qtl-through-contrastive-pca-and-two-way-anova">Identifying stress-responsive Molecular QTL through contrastive PCA and Two-Way ANOVA</h3>
<p>MODAS2 innovatively employs the contrastive PCA algorithm to separate the stress-response effects of natural variation on molecular traits from the genetic effects of natural variation on molecular traits. The separated stress response effects are then used as stress response indices for identifying stress-responsive molecular QTL. After identifying QTL using the stress response indices, the significance of the stress response for these QTL is further assessed through two-way ANOVA, resulting in the final stress-responsive molecular QTL. MODAS2 uses the subcommand <code>contrast</code> to identify stress-responsive molecular QTL. The command line is as follows:</p>
<pre><code class='nohighlight'>MODAS contrast -stress_phe example_data/test_salt.phe.csv -control_phe example_data/test_control.phe.csv -gwas -g example_data/example_geno.contrast -genome_cluster example_data/example_contrast.genome_cluster.csv -p 10 -o example
</code></pre>
<p><code>contrast</code> subcommand generates five files including <code>example.scpca_pc.phe.csv</code>, <code>example.scpca_pc.beta_test.csv</code>, <code>example.scpca_pc.normalized_phe.csv</code>, <code>example.region_gwas_qtl_res.anova.csv</code>, <code>example.region_gwas_qtl_res.anova.sig.csv</code> and <code>example.region_gwas_bad_qtl_res.csv</code>, which contain the stress response indices of molecular traits, statistical test results of the stress response indices, normalized stress response indices of molecular traits, stress response indices QTL results including two-way ANOVA P values, stress-responsive molecular QTL results and unreliable molecular QTL results, respectively.</p>
<h3 id="multi-trait-qtl-colocalization-analysis-based-on-image-matching-algorithms">Multi-Trait QTL Colocalization Analysis Based on Image Matching Algorithms</h3>
<p>QTL colocalization analysis is an effective method for integrating functional information across different traits. However, existing methods rarely perform multi-trait QTL colocalization analysis. MODAS2 employs an image matching algorithm to score the colocalization degree between pairs of QTLs, and then uses clustering algorithms to quickly achieve multi-trait QTL colocalization analysis. Multi-trait QTL colocalization analysis can be performed using the <code>coloc</code> subcommand of MODAS2. The command line is as follows:</p>
<pre><code class='nohighlight'>MODAS coloc -qtl example_data/test_coloc.qtl.csv -g example_data/example_geno.contrast -gwas_dir example_data/gwas_coloc_test/ -p 6 -o example
</code></pre>
<p><code>coloc</code> subcommand generates three files including <code>example.coloc_res.csv</code>, <code>example.coloc_pairwise.csv</code> and <code>example.dis_res.csv</code>, which contain the results of the co-localized QTL clusters, the pairwise QTL co-localization results, and the similarity results between pairwise QTLs, respectively.</p>
<p><strong>Note</strong>: Sample data for MODAS2 can be downloaded via zenodo.</p>
<h2 id="modules-in-modas">Modules in MODAS</h2>
<h3 id="generate-pseudo-genotype-files">Generate pseudo-genotype files</h3>
<p>The pseudo-genotype files can be generated from the <code>-genome_cluster</code> parameter in the subcommand <code>genoidx</code>. Since the genotype files are relatively large, <code>-p</code> is generally added to specify multi-threading to accelerate the generation of pseudo-genotype files.</p>
<pre><code class='nohighlight'>MODAS genoidx -g ./chr_HAMP -genome_cluster -p 10 -o chr_HAMP
</code></pre>
<h3 id="prescreen-candidate-genomic-regions-for-omics-data">Prescreen candidate genomic regions for omics data</h3>
<p>Prescreening of candidate genomic regions for omics data/molecular traits (mTraits) is generated by the subcommand <code>prescreen</code>. In this step, the pseudo-genotype files are first used to perform association analysis on the mTraits through linear model (LM). Then, the significantly associated genomic regions (SAGRs) for each mTrait are re-analyzed through mixed linear model (MLM). Finally, the SAGRs pass the MLM screening are used as the candidate genomic regions.</p>
<pre><code class='nohighlight'>MODAS prescreen -g ./chr_HAMP -genome_cluster ./chr_HAMP.genome_cluster.csv -phe ./E3_log2.normalized_phe.csv -p 20 -o E3_log2
</code></pre>
<h3 id="perform-regional-association-analysis-to-identify-qtls">Perform regional association analysis to identify QTLs</h3>
<p>The QTL identification step is generated by the subcommand <code>regiongwas</code>. In this step, the SNPs from the candidate genomic regions are extracted to perform regional association analysis through MLM, and the QTLs for each mTrait are identified according to the results of SNP-based regional association analysis.</p>
<pre><code class='nohighlight'>MODAS regiongwas -g ./chr_HAMP -phe ./E3_log2.sig_omics_phe.csv -phe_sig_qtl ./E3_log2.phe_sig_qtl.csv -p 20 -o E3_log2
</code></pre>
<h3 id="perform-mendelian-randomization-analysis">Perform Mendelian randomization analysis</h3>
<p>Mendelian randomization (MR) analysis is generated by the subcommand <code>mr</code>. In this step, the peak SNP of a QTL for each trait is used to infer the casual relationship between trait pairs. MR analysis in MODAS can be performed using either linear model (LM) or mixed linear model (MLM), with parameters <code>-lm</code> and <code>-mlm</code>, respectively.</p>
<pre><code class='nohighlight'># lm model
MODAS mr -g chr_HAMP -exposure AMP_kernel_transcriptome_v4_FPKM_correct.sig_eqtl.qqnorm.csv -outcome blup_traits_final.new.csv -qtl AMP_kernel_transcriptome_qtl_res.csv -lm  -o AMP_kernel_transcriptome_MR_lm
# mlm model
MODAS mr -g chr_HAMP -exposure AMP_kernel_transcriptome_v4_FPKM_correct.sig_eqtl.qqnorm.csv -outcome blup_traits_final.new.csv -qtl AMP_kernel_transcriptome_qtl_res.csv -mlm  -o AMP_kernel_transcriptome_MR_lm
</code></pre>
<h3 id="whole-genome-wide-association-analysis-and-visualization">Whole genome-wide association analysis and visualization</h3>
<p>GWAS visualization is generated by the subcommand <code>visual</code>. In this step, MODAS uses specified QTL and trait files as inputs, to perform GWAS and generate whole-genome level Manhattan plots, and displays the results through a HTML based web page.</p>
<pre><code class='nohighlight'># visualization
MODAS visual -g  chr_HAMP -phe E3_log2.normalized_phe.csv -qtl E3_log2.local_gwas_qtl_res.csv -p 6 -visual
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../tutorial/" class="btn btn-neutral float-right" title="Tutorial">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../installation/" class="btn btn-neutral" title="Installation"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../installation/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../tutorial/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../javascripts/config.js" defer></script>
      <script src="../javascripts/mathjax.js" defer></script>
      <script src="https://polyfill.io/v3/polyfill.min.js?features=es6" defer></script>
      <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
